test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1475,6 +1475,11 @@ TEST_F(LuaHttpFilterTest, ImmediateResponse) {
   const uint64_t mem_use_at_start = script_config->runtimeBytesUsed();
 
   uint64_t num_loops = 2000;
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) &&                                    \
+    __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+  // envoy starts crashing on s390x when the number is >= 1300
+  num_loops = 1200;
+#endif
```

Comment:
Why does it crash? That doesn't make a lot of sense to me.

Reply:
Here is the stacktrace for num_loops = 2000:
```
[ RUN      ] LuaHttpFilterTest.ImmediateResponse
TestRandomGenerator running with seed -1327730352
[2022-07-07 15:48:38.693][1078321][critical][backtrace] [./source/server/backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x3ffc3e7f000
[2022-07-07 15:48:38.807][1078321][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-07-07 15:48:38.807][1078321][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: 0/1.20.4/test/DEBUG/OpenSSL
[2022-07-07 15:48:38.834][1078321][critical][backtrace] [./source/server/backtrace.h:96] #0: Envoy::SignalAction::sigHandler() [0x81a78b94]
[2022-07-07 15:48:38.834][1078321][critical][backtrace] [./source/server/backtrace.h:98] #1: [0x3ffb827e490]
[2022-07-07 15:48:38.857][1078321][critical][backtrace] [./source/server/backtrace.h:96] #2: base::subtle::NoBarrier_Load() [0x82bbc8fe]
[2022-07-07 15:48:38.878][1078321][critical][backtrace] [./source/server/backtrace.h:96] #3: base::internal::HookList<>::empty() [0x82bb9970]
[2022-07-07 15:48:38.904][1078321][critical][backtrace] [./source/server/backtrace.h:96] #4: operator delete() [0x82ce314c]
[2022-07-07 15:48:38.928][1078321][critical][backtrace] [./source/server/backtrace.h:96] #5: testing::internal::AnyMatcherImpl<>::~AnyMatcherImpl() [0x8018a1d2]
[2022-07-07 15:48:38.949][1078321][critical][backtrace] [./source/server/backtrace.h:96] #6: std::_Sp_counted_ptr<>::_M_dispose() [0x8018a726]
[2022-07-07 15:48:38.972][1078321][critical][backtrace] [./source/server/backtrace.h:96] #7: std::_Sp_counted_base<>::_M_release() [0x800ae3dc]
[2022-07-07 15:48:38.994][1078321][critical][backtrace] [./source/server/backtrace.h:96] #8: std::__shared_count<>::~__shared_count() [0x800ae2a4]
[2022-07-07 15:48:39.014][1078321][critical][backtrace] [./source/server/backtrace.h:96] #9: std::__shared_ptr<>::~__shared_ptr() [0x8018a814]
[2022-07-07 15:48:39.036][1078321][critical][backtrace] [./source/server/backtrace.h:96] #10: std::shared_ptr<>::~shared_ptr() [0x8018a7e0]
[2022-07-07 15:48:39.056][1078321][critical][backtrace] [./source/server/backtrace.h:96] #11: testing::internal::MatcherBase<>::~MatcherBase() [0x8018a3e4]
[2022-07-07 15:48:39.077][1078321][critical][backtrace] [./source/server/backtrace.h:96] #12: testing::Matcher<>::~Matcher() [0x80189de0]
[2022-07-07 15:48:39.097][1078321][critical][backtrace] [./source/server/backtrace.h:96] #13: testing::internal::TypedExpectation<>::~TypedExpectation() [0x80189f4c]
[2022-07-07 15:48:39.120][1078321][critical][backtrace] [./source/server/backtrace.h:96] #14: testing::internal::TypedExpectation<>::~TypedExpectation() [0x80189fa6]
[2022-07-07 15:48:39.141][1078321][critical][backtrace] [./source/server/backtrace.h:96] #15: std::_Sp_counted_ptr<>::_M_dispose() [0x8018acc6]
[2022-07-07 15:48:39.141][1078321][critical][backtrace] [./source/server/backtrace.h:96] #16: std::_Sp_counted_base<>::_M_release() [0x800ae3dc]
[2022-07-07 15:48:39.141][1078321][critical][backtrace] [./source/server/backtrace.h:96] #17: std::__shared_count<>::~__shared_count() [0x800ae2a4]
[2022-07-07 15:48:39.164][1078321][critical][backtrace] [./source/server/backtrace.h:96] #18: std::__shared_ptr<>::~__shared_ptr() [0x800eb824]
[2022-07-07 15:48:39.193][1078321][critical][backtrace] [./source/server/backtrace.h:96] #19: std::shared_ptr<>::~shared_ptr() [0x800e9070]
[2022-07-07 15:48:39.221][1078321][critical][backtrace] [./source/server/backtrace.h:96] #20: testing::Expectation::~Expectation() [0x82b6b4e0]
[2022-07-07 15:48:39.247][1078321][critical][backtrace] [./source/server/backtrace.h:96] #21: __gnu_cxx::new_allocator<>::destroy<>() [0x80129d86]
[2022-07-07 15:48:39.277][1078321][critical][backtrace] [./source/server/backtrace.h:96] #22: std::allocator_traits<>::destroy<>() [0x80129cec]
[2022-07-07 15:48:39.301][1078321][critical][backtrace] [./source/server/backtrace.h:96] #23: std::_Rb_tree<>::_M_destroy_node() [0x80129c54]
[2022-07-07 15:48:39.323][1078321][critical][backtrace] [./source/server/backtrace.h:96] #24: std::_Rb_tree<>::_M_drop_node() [0x80129be2]
[2022-07-07 15:48:39.346][1078321][critical][backtrace] [./source/server/backtrace.h:96] #25: std::_Rb_tree<>::_M_erase() [0x80129ad0]
[2022-07-07 15:48:39.367][1078321][critical][backtrace] [./source/server/backtrace.h:96] #26: std::_Rb_tree<>::~_Rb_tree() [0x80129a06]
[2022-07-07 15:48:39.389][1078321][critical][backtrace] [./source/server/backtrace.h:96] #27: std::set<>::~set() [0x801299c0]
[2022-07-07 15:48:39.410][1078321][critical][backtrace] [./source/server/backtrace.h:96] #28: testing::ExpectationSet::~ExpectationSet() [0x80128bd0]
[2022-07-07 15:48:39.432][1078321][critical][backtrace] [./source/server/backtrace.h:96] #29: testing::internal::ExpectationBase::~ExpectationBase() [0x82b6776a]
[2022-07-07 15:48:39.455][1078321][critical][backtrace] [./source/server/backtrace.h:96] #30: testing::internal::TypedExpectation<>::~TypedExpectation() [0x800e92c8]
[2022-07-07 15:48:39.479][1078321][critical][backtrace] [./source/server/backtrace.h:96] #31: testing::internal::TypedExpectation<>::~TypedExpectation() [0x800e9306]
[2022-07-07 15:48:39.500][1078321][critical][backtrace] [./source/server/backtrace.h:96] #32: std::_Sp_counted_ptr<>::_M_dispose() [0x800ea1f6]
[2022-07-07 15:48:39.500][1078321][critical][backtrace] [./source/server/backtrace.h:96] #33: std::_Sp_counted_base<>::_M_release() [0x800ae3dc]
[2022-07-07 15:48:39.500][1078321][critical][backtrace] [./source/server/backtrace.h:96] #34: std::__shared_count<>::~__shared_count() [0x800ae2a4]
[2022-07-07 15:48:39.500][1078321][critical][backtrace] [./source/server/backtrace.h:96] #35: std::__shared_ptr<>::~__shared_ptr() [0x800eb824]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #36: std::shared_ptr<>::~shared_ptr() [0x800e9070]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #37: testing::Expectation::~Expectation() [0x82b6b4e0]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #38: __gnu_cxx::new_allocator<>::destroy<>() [0x80129d86]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #39: std::allocator_traits<>::destroy<>() [0x80129cec]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #40: std::_Rb_tree<>::_M_destroy_node() [0x80129c54]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #41: std::_Rb_tree<>::_M_drop_node() [0x80129be2]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #42: std::_Rb_tree<>::_M_erase() [0x80129ad0]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #43: std::_Rb_tree<>::~_Rb_tree() [0x80129a06]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #44: std::set<>::~set() [0x801299c0]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #45: testing::ExpectationSet::~ExpectationSet() [0x80128bd0]
[2022-07-07 15:48:39.501][1078321][critical][backtrace] [./source/server/backtrace.h:96] #46: testing::internal::ExpectationBase::~ExpectationBase() [0x82b6776a]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #47: testing::internal::TypedExpectation<>::~TypedExpectation() [0x80189f68]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #48: testing::internal::TypedExpectation<>::~TypedExpectation() [0x80189fa6]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #49: std::_Sp_counted_ptr<>::_M_dispose() [0x8018acc6]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #50: std::_Sp_counted_base<>::_M_release() [0x800ae3dc]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #51: std::__shared_count<>::~__shared_count() [0x800ae2a4]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #52: std::__shared_ptr<>::~__shared_ptr() [0x800eb824]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #53: std::shared_ptr<>::~shared_ptr() [0x800e9070]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #54: testing::Expectation::~Expectation() [0x82b6b4e0]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #55: __gnu_cxx::new_allocator<>::destroy<>() [0x80129d86]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #56: std::allocator_traits<>::destroy<>() [0x80129cec]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #57: std::_Rb_tree<>::_M_destroy_node() [0x80129c54]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #58: std::_Rb_tree<>::_M_drop_node() [0x80129be2]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #59: std::_Rb_tree<>::_M_erase() [0x80129ad0]
[2022-07-07 15:48:39.523][1078321][critical][backtrace] [./source/server/backtrace.h:96] #60: std::_Rb_tree<>::~_Rb_tree() [0x80129a06]
================================================================================
Target //test/extensions/filters/http/lua:lua_filter_test up-to-date:
  bazel-bin/test/extensions/filters/http/lua/lua_filter_test
INFO: Elapsed time: 144.494s, Critical Path: 144.22s
INFO: 4 processes: 4 processwrapper-sandbox.
INFO: Build completed, 1 test FAILED, 4 total actions
//test/extensions/filters/http/lua:lua_filter_test                       FAILED in 81.5s
```
I don't know yet the root cause of this crash on s390x. I also noticed similar problem for TSAN use-case on x86 https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/http/lua/lua_filter_test.cc#L1478-L1482

Reply:
This seems like a potentially real problem? I think it should be debugged vs. just modifying the test without understanding the cause.

Reply:
Thanks for the review, I'll try to debug.